### ISSUE TITLE
BUG: Specifying dtype argument in pandas.read_csv() throws "ValueError: entry not a 2- or 3- tuple"

### ISSUE THREAD

**Issue by wjs20** [pandas-dev/pandas#43413]
#### Problem description
I am trying to specify different dtypes for different columns of my dataframe at initialization by passing a dictionary of dtypes to the dtype parameter of **pandas.read_csv()**, however it is throwing a **value error** (code and traceback included below).

#### Code example

```python
import pandas as pd, string, random, numpy as np
from itertools import islice, cycle

# create example columns of different datatypes
categories = list(islice(cycle(['red', 'green', 'blue']), 0, 1000))
strings    = [''.join(random.sample(string.ascii_lowercase, 15)) for i in range(1000)]
ints       = [random.randrange(2000) for i in range(1000)]
floats     = [random.random() for i in range(1000)]

data = list(zip(categories, strings, ints, floats))

# specify the desired dtype of the different columns
dtypes = {'categories':'category', 'strings':'string', 'ints':'int32', 'floats':'float64'}

df = pd.DataFrame(data, columns = ['categories', 'strings', 'ints', floats'], dtype=dtypes)
```
#### Traceback

<details>

```python
ValueError                                Traceback (most recent call last)
/tmp/ipykernel_19908/2069478510.py in <module>
     10 
     11 dtypes = {'categories':'category', 'strings':'string', 'ints':'int64', 'floats':'float64'}
---> 12 df = pd.DataFrame(data, columns = ['my_categories', 'my_strings', 'my_ints', 'my_floats'], dtype=dtypes)

~/germlines/germlines_env/lib/python3.8/site-packages/pandas/core/frame.py in __init__(self, data, index, columns, dtype, copy)
    590             data = {}
    591         if dtype is not None:
--> 592             dtype = self._validate_dtype(dtype)
    593 
    594         if isinstance(data, DataFrame):

~/germlines/germlines_env/lib/python3.8/site-packages/pandas/core/generic.py in _validate_dtype(cls, dtype)
    440         """validate the passed dtype"""
    441         if dtype is not None:
--> 442             dtype = pandas_dtype(dtype)
    443 
    444             # a compound dtype

~/germlines/germlines_env/lib/python3.8/site-packages/pandas/core/dtypes/common.py in pandas_dtype(dtype)
   1774     # raise a consistent TypeError if failed
   1775     try:
-> 1776         npdtype = np.dtype(dtype)
   1777     except SyntaxError as err:
   1778         # np.dtype uses `eval` which can raise SyntaxError

~/germlines/germlines_env/lib/python3.8/site-packages/numpy/core/_internal.py in _usefields(adict, align)
     60         names = None
     61     if names is None:
---> 62         names, formats, offsets, titles = _makenames_list(adict, align)
     63     else:
     64         formats = []

~/germlines/germlines_env/lib/python3.8/site-packages/numpy/core/_internal.py in _makenames_list(adict, align)
     30         n = len(obj)
     31         if not isinstance(obj, tuple) or n not in (2, 3):
---> 32             raise ValueError("entry not a 2- or 3- tuple")
     33         if n > 2 and obj[2] == fname:
     34             continue

ValueError: entry not a 2- or 3- tuple
```

</details>

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 5f648bf1706dd75a9ca0d29f26eadfbb595fe52b
python           : 3.8.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.72-microsoft-standard-WSL2
Version          : #1 SMP Wed Oct 28 23:40:43 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.3.2
numpy            : 1.21.2
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 20.0.2
setuptools       : 44.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.1
IPython          : 7.27.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.3
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.7.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

**Comment by phofl** [comment]
There is no read_csv call.

the dataframe constructor accepts only a single dtype

**Comment by wjs20** [comment]
Ah sorry, in my original problem I called .read_csv() - which worked - but for the dummy problem I just constructed a data frame using pd.DataFrame. specifying different dtypes worked for pd.read_csv() so I'm guessing it only works there, and not with the DataFrame constructor? 

What would be the simplest way to specify different datatypes when creating a dataframe from objects in memory rather than from disc.
